File: src/main/java/xyz/capybara/clamav/commands/Ping.java
package xyz.capybara.clamav.commands;

import xyz.capybara.clamav.exceptions.InvalidResponseException;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class Ping extends Command<Void> {

    @Override
    public String getCommandString() {
        return "PING";
    }

    @Override
    protected CommandFormat getFormat() {
        return CommandFormat.NULL_CHAR;
    }

    @Override
    protected Void parseResponse(String responseString) {
        if (!responseString.equals("PONG")) {
            throw new InvalidResponseException(responseString);
        }

        log.info(responseString);

        return null;
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/Ping.kt
package xyz.capybara.clamav.commands

import xyz.capybara.clamav.InvalidResponseException

internal object Ping : Command<Unit>() {
    override val commandString
        get() = "PING"

    override val format
        get() = Command.CommandFormat.NULL_CHAR

    override fun parseResponse(responseString: String) {
        if (responseString != "PONG") {
            throw InvalidResponseException(responseString)
        }
        logger.debug(responseString)
    }
}
